libminifi/include/properties/Properties.h

```diff
@@ -65,7 +65,7 @@ class Properties {
    * @param value value in which to place the map's stored property value
    * @returns true if found, false otherwise.
    */
-  bool get(const std::string &key, std::string &value) const;
+  bool getString(const std::string &key, std::string &value) const;
```

Comment:
This way it is easier to see which function is getting called.  We don't need `get()` to be virtual, and hiding a non-virtual function isn't nice.  Also, Properties has a `getInt()` method, so `getString()` makes sense.

Reply:
> We don't need get() to be virtual, and hiding a non-virtual function isn't nice.

I don't think hiding is a big issue. What I don't like is that a change in a derived class forces the parent class to change in an undesirable way.

> Properties has a getInt() method, so getString() makes sense.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ro-equivalent

Also, getInt isn't too important, just a convenience overload from the times before optional and map. `Properties` is a string key-string value store, so I as a user shouldn't need to say `getString` when retrieving a value, because I already said that by using the `Properties` class.

Reply:
I think that by naming both functions in the parent and child classes `get()`, we would be setting a trap for our future selves, and for anyone who will work on this code.  I agree `getString()` is not a great name, but I would prefer to rename it to anything other than `get()`.  How about `value()`, `getProperty()`, `getPropertyValue()` or something similar?

Reply:
After discussion, we agreed to leave it as is, because all of the alternatives were suboptimal.